contribute-to-tidb/review-a-pr-stage.md

```diff
@@ -0,0 +1,233 @@
+
+Note that comments are different from documentation of classes, modules, or functions, which should instead express the purpose of a piece of code, how it should be used, and how it behaves when used.
+
+### Style
+
+Make sure the PR follows our [code style and quality guide](code-style-and-quality-guide.md). For Go and Rust, there are built in tools with the compiler toolchain.
+
+If you want to improve style points outside the style guide, you can comment that you think it would improve the code readability but isn't mandatory. Don't block PRs from being submitted based only on personal style preferences.
+
+The PR author should not include major style changes combined with other changes. For example, if a PR wants to reformat the whole file should not make functional changes. This has been explained as "One PR just does one thing" in the previous section.
+
+### Consistency
+
+What if the existing code is inconsistent with the style guide? The style guide is the absolute authority: if something is required by the style guide, the PR should follow the guidelines.
+
+In some cases, the style guide makes recommendations rather than declaring requirements. In these cases, it’s a judgment call whether the new code should be consistent with the recommendations or the surrounding code. Bias towards following the style guide unless the local inconsistency would be too confusing
+
+If no other rule applies, the author should maintain consistency with the existing code.
+
+Either way, encourage the author to file a bug and add a TODO for cleaning up existing code.
+
+### Documentation
+
+If a PR changes how users build, test, interact with, or release code, the reviewer are supposed to check whether it also updates related documentation, such as READMEs, and any generated reference docs. 
+
+If a PR deletes or deprecates code, it should consider whether the documentation should also be deleted.
+
+If documentation is missing, the reviewer should ask for it.
+
+#### Error Handling
+
+Be careful with error or exception handling. This is one of the most common places for bugs.
+
+* Does the test cover the error code path?
+* Does a panic make the whole process exit for server services?
+* Does the code handle the resource releasing properly?
+* Especially, for code including concurrent goroutines, is there a goroutine leak or memory leak after the error?
+
+Sometimes it is difficult for a real program to run into the error code path, and even more difficult to reproduce and debug if it really happens. Then it is necessary to mock the error code path. Tools like [failpoint](https://github.com/pingcap/failpoint) are available for that purpose.
+
+### Performance
+
+Does the PR impose any performance impact? Sometimes a minor change on the hot path that seems unrelated may still affect performance. Is the algorithm in the PR suitable for that case?
```

Comment:
Does the PR impact performance? Sometimes, a seemingly minor, unrelated change on the hot may still affect performance.

Reply:
```suggestion
Does the PR impact performance? Sometimes, a seemingly minor, unrelated change on the hot may still affect performance.

Is the algorithm in the PR suitable for that case?
```